src/org/opendatakit/briefcase/ui/pull/PullPanel.java

```diff
@@ -52,13 +53,14 @@
   private TerminationFuture terminationFuture;
   private Optional<Source<?>> source = Optional.empty();
 
-  public PullPanel(PullPanelForm view, PullForms forms, BriefcasePreferences tabPreferences, BriefcasePreferences appPreferences, TerminationFuture terminationFuture) {
+  public PullPanel(PullPanelForm view, PullForms forms, BriefcasePreferences tabPreferences, BriefcasePreferences appPreferences, TerminationFuture terminationFuture, Analytics analytics) {
```

Comment:
I think I’d rather see `ComponentListener analyticsComponentListener` than the whole `Analytics` object passed in, but not a big deal.

Reply:
OK, we can discuss that after getting rid of all implicit coupling on global artifacts. There's still hidden stuff around... 

I'm for passing each top-level artifact the least number of dependencies on their minimal possible form.